DataFormats/PatCandidates/src/PackedCandidate.cc

```diff
@@ -62,9 +62,12 @@ void pat::PackedCandidate::packVtx(bool unpackAfterwards) {
 }
 
 void pat::PackedCandidate::unpack() const {
-    p4_ = PolarLorentzVector(MiniFloatConverter::float16to32(packedPt_),
+    float pt = MiniFloatConverter::float16to32(packedPt_);
+    double shift = (pt<1. ? 0.1*pt : 0.1/pt); // shift particle phi to break degeneracies in angular separations
+    double phi = int16_t(packedPhi_)*3.2f/std::numeric_limits<int16_t>::max() + shift*3.2/std::numeric_limits<int16_t>::max();
```

Comment:
OK, I tried to suggest something stable under pack unpack operations.
Smoother transformation could always be picked up.

The feedback on this PR so far sounds like the PR is a no go anyways.
On Jun 11, 2015 10:18 PM, "Dinko Ferencek" <notifications@github.com> wrote:

> In DataFormats/PatCandidates/src/PackedCandidate.cc
> <https://github.com/cms-sw/cmssw/pull/9554#discussion_r32288004>:
>
> > @@ -62,9 +62,12 @@ void pat::PackedCandidate::packVtx(bool unpackAfterwards) {
> >  }
> >
> >  void pat::PackedCandidate::unpack() const {
> > -    p4_ = PolarLorentzVector(MiniFloatConverter::float16to32(packedPt_),
> > +    float pt = MiniFloatConverter::float16to32(packedPt_);
> > +    double shift = (pt<1. ? 0.1*pt : 0.1/pt); // shift particle phi to break degeneracies in angular separations
> > +    double phi = int16_t(packedPhi_)*3.2f/std::numeric_limits<int16_t>::max() + shift*3.2/std::numeric_limits<int16_t>::max();
>
> @slava <https://github.com/slava>, even if we were packing in theta, the
> grid would still exist in eta. Granted, it would not be uniform in eta, but
> degeneracies would still be possible. Furthermore, even if at the same time
> we had a non-linear packing of phi, there would be some symmetry among the
> 4 quadrants in the eta-phi plane and degeneracies would in principle still
> be possible (although I would assume they would be significantly less
> likely than now).
>
> I don't think it makes sense to pack in cos(phi) since the measured phi
> resolution should be more or less invariant under phi. Hence, packing in
> phi is what makes most sense. Concerning packing in theta, yes, it probably
> makes more sense from the experimental point of view. However, constructors
> of 4-vector classes generally do not take theta as input which means that
> packing and unpacking would involve computationally more intensive
> non-linear functions. So here again I think it makes more sense to pack eta
> directly. In fact, the grid in eta becomes in some sense too dense in the
> forward region so there is no real issue.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/9554/files#r32288004>.
>